### PR TITLE
Display actual usernames instead of UIDs in Event/Serie details

### LINKED ISSUE #270: Resolve author display name for Event/Serie details
Replace the hard-coded/UID fallback with the actual display name fetched from the user profile store when opening EventDetails or SerieDetails.
Why: Current UI shows a raw Firebase UID because we never implemented the username lookup.
Scope: ShowEventScreen + SerieDetailsScreen + ViewModels + repository call to fetch display name.

### PR DESCRIPTION
**Pull request by BryanDMP**
## Description
Resolves the issue where Event and Serie detail screens displayed raw Firebase UIDs instead of user display names for event/serie owners.

## Screenshots

| Event details | Serie details |
|:---:|:---:|
| Event details | Serie details |

## Changes
- **ShowEventViewModel**: Fetch owner username from `ProfileRepository` in `loadEvent()`
- **SerieDetailsViewModel**: Implement `getOwnerDisplayName()` to fetch username from `ProfileRepository`
- **Tests**: Added proper mocking of `ProfileRepository` in all ViewModel and Screen tests
- **Tests**: Added coverage for edge cases (profile not found, repository errors, empty owner IDs)

## Testing
-  All existing tests updated to mock `ProfileRepository`
-  New tests for owner name display with various scenarios
-  Verified "Created by [Username]" displays correctly
-  Fallback to "UNKNOWN" when profile not found

## Related
Closes #270 - Resolve author display name for Event/Serie details

## AI-assistance
The test files were adapted and refactored with the assistance of AI tools.

### CODE REVIEW

File: app/src/main/java/com/android/joinme/ui/overview/SerieDetailsScreen.kt [hunk 1]
```diff
@@ -23,77 +24,26 @@ import com.google.firebase.auth.ktx.auth
 import com.google.firebase.ktx.Firebase
 import kotlinx.coroutines.launch
 
-/**
- * Test tags for UI testing of the Serie Details screen components.
- *
- * Provides consistent identifiers for testing individual UI elements.
- */
```
[the remark this answers is not in the document]
Comment (BryanDMP) [reply app/src/main/java/com/android/joinme/ui/overview/SerieDetailsScreen.kt]: Yep, my bad that was accidental!

**Review by Saifurrhn — APPROVED** [review]
Not much to say, after going through the branch, all seems to work fine. Good job!
